src/test/java/com/google/devtools/build/lib/rules/objc/ObjcLibraryTest.java

```diff
@@ -380,7 +380,7 @@ public void testProvidesObjcHeadersWithDotMFiles() throws Exception {
     assertThat(getArifactPaths(target, HEADER))
         .containsExactly("objc/a.h", "objc/b.h", "objc/f.m");
     assertThat(getArifactPaths(depender, HEADER))
-        .containsExactly("objc/a.h", "objc/b.h", "objc/f.m", "objc2/d.h", "objc2/e.m");
+        .containsExactly("objc/a.h", "objc/b.h", "objc/f.m", "objc/private.h", "objc2/d.h", "objc2/e.m");
```

Comment:
This is checking the `header` field of the `ObjcProvider`, right? AFAIK, it definitely should _not_ be propagating private headers that are only listed in `srcs`.

Reply:
I was going by [this comment](https://github.com/bazelbuild/bazel/pull/5954#issuecomment-417259877) from #5954.